### PR TITLE
Create statsmodels entry for Python

### LINKED ISSUE #5637: [Concept Entry] Python : statsmodels
### Reason/inspiration (optional)

We would like a new entry on the `statsmodels` concept under Python. The entry should go in a new file under `docs/content/python/concepts/statsmodels/statsmodels.md`.

The entry should include:  
  
- An introduction to the concept  
- A `Syntax` section that provides the syntax for the concept  
- An `Example` section that provides an example demonstrating the concept in use  
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use  
  
Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by arisdelacruz**
### Description

An attempt to create statsmodels entry for Python

### Issue Solved

Closes #5637 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/python/concepts/statsmodels/statsmodels.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+---
+Title: 'Statsmodels'
+Description: 'The statsmodels library is used to estimate statistical models and perform statistical tests in Python.'
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:3]: We should rephrase it just a little to make it clearer and easier to understand.

```suggestion
Description: 'The statsmodels library is used to estimate the statistical models and perform statistical tests.' 
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
+---
+Title: 'Statsmodels'
+Description: 'The statsmodels library is used to estimate statistical models and perform statistical tests in Python.' 
+Subjects:
+  - 'Data Science'
+  - 'Statistics'
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:6]: `Statistics` is not a valid subject. Please refer to the list of subjects in [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md).

```suggestion
  - 'Data Visualization'
  - 'Computer Science'
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+Tags:
+  - 'Python'
+  - 'Statistics'
+  - 'Statsmodels'
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:10]: `Statsmodels` is not a valig tag. Please refer to the list of tags in [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).

```suggestion
  - 'Data'
  - 'Models'
  - 'Datasets'
  - 'Machine Learning'
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+In Python, the `statsmodels` library is used to estimate the statistical models, and perform statistical tests. It is built on top of `numpy`, `scipy`, and `pandas`. It provides classes and functions for the estimation of many different statistical models, as well as for conducting statistical tests, and statistical data exploration.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:16]: The last sentence is just repeating what the first sentence already states. We can remove it.

```suggestion
In Python, the `statsmodels` library is used to estimate the statistical models and perform statistical tests. It is built on top of `numpy`, `scipy`, and `pandas`. 
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+The same is widely used in the field of econometrics, and it is also used in other fields such as finance, marketing, and social sciences. It is a powerful tool for statistical analysis, and it provides a wide range of statistical models and tests.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:18]: This sentence can be simplified as follows, avoiding repetition - 

```suggestion
It is widely used in econometrics and other fields such as finance, marketing, and social sciences.
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+It provides classes and functions for the estimation of many different statistical models, such as linear regression, generalized linear models, time series analysis, and more. It also provides functions for conducting statistical tests, such as hypothesis tests, and for statistical data exploration.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:20]: Again, this mostly repeats what we have already told the learner. We can simplify this as - 

```suggestion
It supports various models, including linear regression, generalized linear models, time series analysis, and more.
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+- `Estimation of statistical models`: provides classes and functions for the estimation of many different statistical models, such as linear regression, generalized linear models, time series analysis, and more.
+
+- `Statistical tests`: provides functions for conducting statistical tests, such as hypothesis tests, and for statistical data exploration.
+
+- `Data exploration`: provides functions for exploring and analyzing data, such as summary statistics, correlation analysis, and more.
+
+- `Visualization`: provides functions for visualizing data, such as scatter plots, histograms, and more.
+
+- `Integration with other libraries`: built on top of `numpy`, `scipy`, and `pandas`, and it integrates well with other libraries in the Python ecosystem.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:32]: Since the list headers are new terms and do not contain coding methods or functions, we can highlight them in bold rather than backticks. Also, the list sentences should begin with capital letters.

```suggestion
- **Estimation of statistical models**: Provides classes and functions for estimating many different statistical models, such as linear regression, generalized linear models, time series analysis, and more.

- **Statistical tests**: Provides functions for conducting statistical tests, such as hypothesis tests.
- **Data exploration**: Provides functions for exploring and analyzing data, such as summary statistics, correlation analysis, and more.

- **Visualization**: Provides functions for visualizing data, such as scatter plots, histograms, and more.

- **Integration with other libraries**: Built on top of `numpy`, `scipy`, and `pandas`, it integrates well with other libraries in the Python ecosystem.
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+## Installation
+
+You can install the `statsmodels` library using `pip` by running the following recommended command:
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:36]: As per the codeacademy [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md), we should avoid using first and second-person pronouns, i.e. "I", "we", and "you".

```suggestion
The `statsmodels` library can be installed using pip by running the following command:
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+```shell
+pip install statsmodels
+```
+
+## Usage
+
+To use the `statsmodels` library in your Python code, you need to import the necessary modules and classes.
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:44]: As per the codeacademy [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md), we should avoid using first and second-person pronouns, i.e. "I", "we", and "you".

```suggestion
To use the `statsmodels` library in Python code, import the necessary modules and classes.
```

File: content/python/concepts/statsmodels/statsmodels.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+y = 2 + 3 * X + np.random.randn(100, 1)
+
+# Fit the linear regression model
+X = sm.add_constant(X)
+model = sm.OLS(y, X)
+results = model.fit()
+
+# Print the summary of the model
+print(results.summary())
+```
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:68]: Add a small explanation as well as the output for the code example to make it easy for the learner to understand.

File: content/python/concepts/statsmodels/statsmodels.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        'C': np.random.randint(0, 2, 100)}
+df = pd.DataFrame(data)
+
+# Fit the ANOVA model
+model = smf.ols('A ~ B + C', data=df)
+results = model.fit()
+
+# Print the summary of the model
+print(results.summary())
+```
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:90]: Add a small explanation as well as the output for the code example to make it easy for learner to understand.

File: content/python/concepts/statsmodels/statsmodels.md [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+np.random.seed(0)
+data = np.random
+
+# Fit the ARIMA model
+model = tsa.ARIMA(data, order=(1, 1, 1))
+results = model.fit()
+
+# Print the summary of the model
+print(results.summary())
+```
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:109]: Add a small explanation as well as the output for the code example to make it easy for learner to understand.

File: content/python/concepts/statsmodels/statsmodels.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+
+# Fit the linear regression model
+X = sm.add_constant(X)
+model = sm.OLS(y, X)
+results = model.fit()
+
+# Perform the hypothesis test
+t_test = results.t_test([0, 1])
+print(t_test)
+```
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:131]: Add a small explanation as well as the output for the code example to make it easy for learner to understand.

File: content/python/concepts/statsmodels/statsmodels.md [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+import statsmodels.api as sm
+import statsmodels.graphics.api as smg
+
+# Generate some random data
+np.random.seed(0)
+X = np.random.rand(10, 10)
+
+# Plot the heatmap
+smg.plot_heatmap(X)
+```
```
Comment (PragatiVerma18) [inline content/python/concepts/statsmodels/statsmodels.md:146]: Add a small explanation as well as the output for the code example to make it easy for learner to understand.

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @arisdelacruz, thanks for contributing this entry. I have some suggestions, please check the comments and do the required changes. Do let me know if you have any questions.

**Comment by PragatiVerma18** [comment]
Hey @arisdelacruz, thanks for committing the suggestions, we are just waiting for the brief explanation and outputs for the examples included, then, this would be good for a second review.

**Comment by arisdelacruz** [comment]
@PragatiVerma18 

Just did as requested. I am not that confident though if I hit it right.

Thanks

**Review by PragatiVerma18 — APPROVED** [review]
Hey @arisdelacruz, thanks for doing the suggested changes. It looks good for a second review to me.

**Review by Sriparno08 — APPROVED** [review]
Looks good, @arisdelacruz!
